src/compiler/mod.rs

```diff
@@ -972,12 +972,57 @@ impl Compiler {
         }
     }
 
+    fn get_llvm_type_with_function_args(&self, typ: parser::Type, param_types: &[llvm::Type], return_type: llvm::Type) -> llvm::Type {
+        match typ {
+            parser::Type::Vector => self.context.double_type().pointer_type(0),
+            parser::Type::Numeric => self.context.double_type(),
+            parser::Type::Function => self
+                .context
+                .function_type(return_type, param_types, false)
+                .pointer_type(0),
+            parser::Type::Null => self.context.void_type(),
+            parser::Type::Ptr => self.context.void_type().pointer_type(0),
+            parser::Type::String => self.context.i8_type().pointer_type(0),
+        }
+    }
+
+    fn get_lambda_func_args(&self, arg: &parser::Param) -> Vec<llvm::Type> {
+        let types: Vec<llvm::Type> = arg
+            .generic_params
+            .iter()
+            .map(|arg2| self.get_llvm_type(arg2.typ))
+            .collect();
+
+        if types.len() > 0  {
+            let len = types.len() - 1;
+            return types[..len].to_vec();
+        } else {
+            return types
+        }
+    }
+
+    fn get_lambda_func_return_type(&self, arg: &parser::Param) -> llvm::Type {
+        let types: Vec<llvm::Type> = arg
+            .generic_params
+            .iter()
+            .map(|arg2| self.get_llvm_type(arg2.typ))
+            .collect();
+
+        if types.len() >= 1  {
+            let len = types.len() - 1;
+            return types[len]
+        } else {
+            return self.context.void_type()
```

Comment:
i would rather have it so that function generic type needs to have return type specfified - so this case would be unreachable then, maybe we can just put unreachable! macro call here